### PR TITLE
fix: Move viewLifeCycleOwner reference to onCreateView

### LINKED ISSUE #2482: App crash on clicking skip
App crashes on pick a city screen and `Skip` is clicked. On opening again, home screen loads but crashes on clicking search or login

Basically, app is in unusable state

### PR DESCRIPTION
**Pull request by tronku**
Fixes #2482 

Changes:  Instead of observing in onCreate method, I have observed the changes in onCreateView as the views get initialized after that in fragments.

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-android-general/pullRequest?prid=4650690&bid=15572851)

File: app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt [hunk 1]
```diff
@@ -50,6 +50,12 @@ class AuthFragment : Fragment(), ComplexBackPressFragment {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt]: Codacy Issue found: [LongMethod - 61/60 - [onCreateView] at /src/app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt:55:5](https://app.codacy.com/app/fossasia/open-event-android-general/pullRequest?prid=4650690)

Comment (tronku) [reply app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt]: onCreateView is the pre-defined functions with a fixed no of parameters. Codacy is giving error "Methods should have one responsibility. Long methods can indicate that a method handles too many cases at once."

File: app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt [hunk 2]
```diff
@@ -50,6 +50,12 @@ class AuthFragment : Fragment(), ComplexBackPressFragment {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+    }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt]: onCreate is redundant. Remove it

**Comment by iamareebjamal** [comment]
Search for other onCreates which might be referring to viewLifecycleOwner

File: app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt (hunk 2 above)
Comment (tronku) [reply app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt]: okayy

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-android-general/pullRequest?prid=4650690&bid=15572851)

File: app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt [hunk 3]
```diff
@@ -48,8 +48,10 @@ class AuthFragment : Fragment(), ComplexBackPressFragment {
     private val safeArgs: AuthFragmentArgs by navArgs()
     private val smartAuthViewModel by sharedViewModel<SmartAuthViewModel>()
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt]: Codacy Issue found: [LongMethod - 61/60 - [onCreateView] at /src/app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt:55:5](https://app.codacy.com/app/fossasia/open-event-android-general/pullRequest?prid=4650690)

Comment (tronku) [reply app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt]: How can I change the pre-defined function of a fragment?

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/fossasia/open-event-android-general/pullRequest?prid=4650690&bid=15572851)

File: app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt (hunk 3 above)
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt]: Codacy Issue found: [LongMethod - 61/60 - [onCreateView] at /src/app/src/main/java/org/fossasia/openevent/general/auth/AuthFragment.kt:55:5](https://app.codacy.com/app/fossasia/open-event-android-general/pullRequest?prid=4650690)

**Pull request merged**